### PR TITLE
(PA-2839) add support for windowsfips

### PR DESCRIPTION
**Pull request by gimmyxd**
TODO:
- [x] Run puppet-agent acceptance suite on a subset of windows boxes

### CODE REVIEW

**Comment by gimmyxd** [comment]
Build some packages for windows here: http://builds.delivery.puppetlabs.net/puppet-agent/8ba57da9ddb5da941a7f782023c3384dfcaf1505/artifacts/?C=M&O=A

```
:platform_data:
  windowsfips-2012-x64:
    :artifact: "./windowsfips/puppet-agent-5.5.7.732.g8ba57da-x64.msi"
    :repo_config: 
  windows-2012-x64:
    :artifact: "./windows/puppet-agent-5.5.7.732.g8ba57da-x64.msi"
    :repo_config: 
  windows-2012-x86:
    :artifact: "./windows/puppet-agent-5.5.7.732.g8ba57da-x86.msi"
    :repo_config: 
```

File: lib/packaging/paths.rb [hunk 1]
```diff
@@ -25,13 +25,15 @@ def arch_from_artifact_path(platform, version, path)
   # with the artifact and path
   def tag_from_artifact_path(path)
     platform = Pkg::Platforms.supported_platforms.find { |p| path =~ /(\/|\.)#{p}[^\.]/ }
+    platform = 'windowsfips' if path =~ /windowsfips/
```
Comment (gimmyxd) [inline lib/packaging/paths.rb]: This is ugly but there the regex above seems to be looking in different places of the `path` string that gets really confusing. So adding something specific for `windowsfips` there just makes it worse.

**Comment by gimmyxd** [comment]
Seems that there are still some parts to be set, i couldn't get it install the right package: https://jenkins-master-prod-1.delivery.puppetlabs.net/view/Adhoc/job/platform_puppet-agent-extra_puppet-agent-integration-suite_adhoc-ad_hoc/RMM_COMPONENT_TO_TEST_NAME=puppet_agent,SLAVE_LABEL=beaker,TEST_TARGET=windowsfips2012r2-64a/669/console

```
Fetching: http://builds.delivery.puppetlabs.net/puppet-agent/2ef96e2183e74db9037669532559212497346215/artifacts/2ef96e2183e74db9037669532559212497346215.yaml
18:12:00     and saving to /tmp/d20190930-2570-ilzw3c/2ef96e2183e74db9037669532559212497346215.yaml
18:12:00   Found packaging platform overrides: {}
18:12:00   Platforms available for this build:
18:12:00   ["windowsfips-2012-x64", "ubuntu-16.04-amd64", "windows-2012-x64", "windows-2012-x86", "el-7-x86_64"]
18:12:00   PLATFORM SPECIFIC INFO for e2oji1oxt3vkp6y.delivery.puppetlabs.net (packaging name 'windowsfips-2012r2-x64'):
18:12:00   - , isnil? true
18:12:00   Warning: 
18:12:00                 :packaging_platform 'windowsfips-2012r2-x64' for host 'e2oji1oxt3vkp6y.delivery.puppetlabs.net' not in build details
18:12:00                 :packaging_platform should be the platform-specific key from this list:
18:12:00                   ["windowsfips-2012-x64", "ubuntu-16.04-amd64", "windows-2012-x64", "windows-2012-x86", "el-7-x86_64"]
18:12:00   
18:12:00   
```

**Comment by gimmyxd** [comment]
i think i figured it out where it comes from: https://github.com/puppetlabs/beaker-hostgenerator/blob/master/lib/beaker-hostgenerator/data.rb#L1438
this should be `windowsfips-2012-x64`

**Comment by gimmyxd** [comment]
Seems to work well together with: https://github.com/gimmyxd/beaker-hostgenerator/commit/4843c64c18a40ddd12cc4037c6a3e373c35cfa8c and https://github.com/gimmyxd/puppet-agent/commit/2ef96e2183e74db9037669532559212497346215

https://jenkins-master-prod-1.delivery.puppetlabs.net/view/Adhoc/job/platform_puppet-agent-extra_puppet-agent-integration-suite_adhoc-ad_hoc/670/

File: spec/lib/packaging/artifactory_spec.rb [hunk 2]
```diff
@@ -26,6 +26,11 @@
         :repo_config => '',
         :additional_artifacts => ["./windows/puppet-agent-extras-5.3.1.34-x86.msi"],
       },
+      'windowsfips-2012-x64' => {
+        :artifact => "./windows/puppet-agent-5.3.1.34-x64.msi",
```
Comment (mwaggett) [inline spec/lib/packaging/artifactory_spec.rb]: Shouldn't this go in a `windowsfips` directory instead of `windows`?

**Pull request merged**